target/linux/ath79/dts/qca9563_tplink_tl-wpa8630p-v2-de.dts

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "qca9563_tplink_tl-wpa8630p-v2.dtsi"
+
+/ {
+	compatible = "tplink,tl-wpa8630p-v2-de", "qca,qca9563";
+	model = "TP-Link WPA8630P v2 (DE)";
+};
+
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "factory-uboot";
+				reg = <0x000000 0x020000>;
+				read-only;
+			};
+
+			partition@20000 {
+				label = "u-boot";
+				reg = <0x020000 0x020000>;
+				read-only;
+			};
+
+			partition@40000 {
+				compatible = "tplink,firmware";
+				label = "firmware";
+				reg = <0x040000 0x5e0000>;
+			};
+
+			partition@620000 {
+				label = "partition-table";
+				reg = <0x620000 0x010000>;
+				read-only;
+			};
+
+			partition@630000 {
+				label = "tplink";
+				reg = <0x630000 0x020000>;
+				read-only;
+			};
+
+			partition@650000 {
+				label = "config";
+				reg = <0x650000 0x020000>;
+				read-only;
+			};
+
+			partition@670000 {
+				label = "plc";
+				reg = <0x670000 0x100000>;
+				read-only;
+			};
+
+			partition@770000 {
+				label = "tplink";
```

Comment:
That's a second partition with the same name.
But anyway, I'd just do the following:

```
			partition@630000 {
				label = "tplink";
				reg = <0x630000 0x1b0000>;
				read-only;
			};
```

Don't think we need anything from this range in OpenWrt, just the "mac" partition should be separate then.

Reply:
You may also call that whole thing "config" if you prefer ...

Reply:
My initial intention was that the "plc" partition might be useful, because it contains the firmware files for the PowerLine SoC. However, they are not needed for operation because the PLC has a flash IC to store the firmware. Taking this into consideration, I agree with your suggestion (also the suggestions below, I'll have to go through the layouts thoroughly once more).